### DISCUSSION TITLE
React js

### DISCUSSION THREAD

**Discussion by Dhaneswar-setha** [community/community#113656] [Programming Help] [answered]
### Hi everyone,

I wanted to start a discussion about React.js and its significance in modern web development. React has become one of the most popular JavaScript libraries for building user interfaces, and for good reason. Its component-based architecture, virtual DOM, and declarative syntax make it a powerful tool for creating dynamic and interactive web applications.

I'd love to hear about your experiences with React.js. What projects have you used it for? What are some of the challenges you've faced, and how did you overcome them? Are there any tips or best practices you'd like to share with the community?

Additionally, let's discuss the latest trends and advancements in the React ecosystem. From state management solutions like Redux and MobX to server-side rendering with Next.js, there's always something new to explore.

Feel free to share your thoughts, ask questions, or start conversations around any aspect of React.js. Let's learn from each other and continue pushing the boundaries of web development with React!

Looking forward to hearing from you all!

**Comment by samus-aran** [comment]
Welcome to the GitHub Community, @Dhaneswar-setha , we're happy you're here! You are more likely to get a useful response if you are posting your question(s) in the applicable category and are explicit about what your project entails--giving a few more details might help someone give you a nudge in the right direction. I've gone ahead and moved it for you. Good luck!

**Comment by KavishkaSashmitha** [comment]
Hello everyone,

I'm diving into React.js, a powerful library reshaping modern web development. As a student navigating between beginner and intermediate levels, I've found React's component-based architecture and declarative syntax to be game-changers.

I've worked on various projects, from simple websites to more complex applications, leveraging React's modularity and reusability. Managing state has been a challenge, but through documentation and community support, I've learned to navigate it.And Also currently working on a quit complex one.

I'm excited to hear about your experiences with React and any tips you have for fellow learners. Let's discuss and grow together!

Best regards,
kavishka

**Comment by synasapmob** [comment, marked as answer]
i worked with React.js in 1.5 years, i think it's as the same Javascript but have much ways to handler
the important you need is background

**Comment by jfullstackdev** [comment]
it's very popular even in the Philippines or abroad but we still use PHP Laravel, particularly for backend

coming from a traditional fetching of data from server once there are updates, it's really hard and confusing to implement React state variables because when for frontend & backend, you are dealing with local state variables that can be passed as props for so many layers of components hence the term `prop drilling`, that may not match or update the data on BE also

the syntax is really not that appealing at first as it is a mixed of JavaScript and HTML even if someone is very familiar with plain JavaScript

but yes, once you get the hang of it, in the end, it's just a tool for you to build websites, the only choice is up to you or your company though React JS is really for medium and big projects, I would prefer Vue.js as it's simplicity
